src/lib/staged_ledger/staged_ledger.ml

```diff
@@ -1215,7 +1215,67 @@ module T = struct
       =
     (List.map ~f:(With_status.map ~f:Transaction.forget) a, b, c, d)
 
-  let check_commands ledger ~verifier (cs : User_command.t With_status.t list) =
+  type transaction_pool_proxy =
```

Comment:
Another ask: could we move all of the new code that was added to `staged_ledger.ml` to a separate `check_commands.ml` in the same package? All new code and `check_commands` function as well?

Motivation: `staged_ledger.ml` is already a big file. I'd prefer that we later split it up. Let's start by not adding more lines to it in new PRs :D